samples/assistant/README.md

```diff
@@ -0,0 +1,221 @@
+Custom skills are useful ways to extend the functionality of an AI assistant. For example, you can create a custom skill that saves a todo item to a database, or a custom skill that queries a database for a list of todo items.
+In both cases, the skills are defined in your app and any the language model doesn't need to know anything about the skill implementation, making it useful for interacting with private data.
+
+This OpenAI extension internally uses the [function calling](https://platform.openai.com/docs/guides/function-calling) functionality available in
+`gpt-3.5-turbo` and `gpt-4` models to implement assistant skills.
+
+## Defining skills
+
+You can define a skill by creating a function that uses the `AssistantSkillTrigger` binding. The following C# in-proc example shows a skill that adds a todo item to a database:
+
+```csharp
+[FunctionName(nameof(AddTodo))]
+public Task AddTodo([AssistantSkillTrigger("Create a new todo task")] string taskDescription)
+{
+    string todoId = Guid.NewGuid().ToString()[..6];
+    return this.todoManager.AddTodoAsync(new TodoItem(todoId, taskDescription));
+}
+```
+
+The `AssistantSkillTrigger` attribute requires a `FunctionDescription` string value, which is text describing what the function does.
+This is critical for the AI assistant to be able to invoke the skill at the right time.
+The name of the function parameter (e.g., `taskDescription`) is also an important hint to the AI assistant about what kind of information to provide to the skill.
+
+> **NOTE:** The `AssistantSkillTrigger` attribute only supports primitive types as function parameters, such as `string`, `int`, `bool`, etc. Function return values can be of any JSON-serializable type.
+
+Any function that uses the `AssistantSkillTrigger` binding will be automatically registered as a skill that can be invoked by any AI assistant.
+The assistant will invoke a skill function whenever it decides to do so to satisfy a user prompt, and will provide function parameters based on
+the context of the conversation. The skill function can then return a response to the assistant, which will be used to satisfy the user prompt.
+
+## Prerequisites
+
+The sample is available in the following language stacks:
+
+* [C# on the in-process worker](csharp-inproc)
+
+You must have the following installed on your local machine in order to run these samples.
+
+* [.NET 6 SDK](https://dotnet.microsoft.com/download/dotnet/6.0) or newer
+* [Azure Functions Core Tools](https://learn.microsoft.com/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash) v4.0.5455 or newer
+* An [OpenAI API key](https://platform.openai.com/account/api-keys) saved into a `OPENAI_API_KEY` environment variable
+* Azure Storage emulator such as [Azurite](https://learn.microsoft.com/azure/storage/common/storage-use-azurite) running in the background
```

Comment:
Root level read me can be referenced here.

As we also want to share the keys for Azure settings.

Reply:
Ah, yes, will reference the root-level README.